FINERACT-783 Fineract on Kubernetes [WIP]

xurror opened the pull request:
minikube configuration files for fineract-server and fineractmysql

https://issues.apache.org/jira/browse/FINERACT-783

## Description
Describe the changes made and why they were made. Ignore if these details are present on the associated Jira ticket

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Commit message starts with the issue number from https://issues.apache.org/jira/projects/FINERACT/. Ex: FINERACT-646 Pockets API.

- [ ] Coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions have been followed.

- [ ] API documentation at https://github.com/apache/fineract/blob/develop/api-docs/apiLive.htm has been updated with details of any API changes.

- [ ] Integration tests have been created/updated for verifying the changes made.

- [ ] All Integrations tests are passing with the new commits.

- [ ] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the list.)

Our guidelines for code reviews is at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide

vorburger commented on kubernetes/Dockerfile:
```diff
@@ -0,0 +1,25 @@
+COPY fineract-db/docker/01-databases.sql /docker-entrypoint-initdb.d/
+RUN ["/usr/local/bin/docker-entrypoint.sh", "mysqld", "--datadir", "/initialized-db"]
+FROM mysql:5.7
+COPY --from=builder /initialized-db /var/lib/mysql
```
Erm... what is this `Dockerfile` for?  You don't seem to be actually using this - given that your `kubernetes/fineract-server-deployment.yaml` actually just uses our standard `image: apache/fineract:latest` - which is great, and how IMHO it should be. Would you perhaps like to simply completely remove this?

xurror replied:
Oh thanks alot for this, will try some couple more things and push

vorburger commented on kubernetes/fineractmysql-deployment.yaml:
```diff
@@ -0,0 +1,101 @@
+      - env:
+        - name: MYSQL_ROOT_PASSWORD
+          value: mysql
+        image: xurror/fineractmysql:1.0
```
this (`image: xurror/fineractmysql:1.0`) is NOK... I've found your https://hub.docker.com/r/xurror/fineractmysql, but we cannot depend on that, this is No Go e.g. from both a security perspective (we have no sources and no idea how you built that container... it could be a anything!), and for future maintenance by others. Why not simply use a standard mysql, or even better mariadb, container?  For example https://hub.docker.com/_/mariadb/ ?

mohitsinha replied:
Do we really need a mysql deployment?

vorburger reviewed (changes requested):
@xurror thank you very much for working on this! I have reviewed this, and found at least 2 things which would have to be addressed for this to be merged.  Also, could I point out (my) #648 to you - I believe you'll find that of interest and useful for what you are doing here as well?

vorburger reviewed (changes requested):
Just marking this as Requested Changes (for Dashboard)

vorburger commented:
xurror@ do you think you'll have time to work on finishing this in the coming weeks? 

It would be a really really cool contribution adding a lot of value to this project!

xurror commented:
Yeah, totally
It will be done before Christmas

On Thu, Dec 12, 2019, 8:49 AM Michael Vorburger ⛑️ <notifications@github.com<mailto:notifications@github.com>> wrote:

xurror@ do you think you'll have time to work on finishing this in the coming weeks?

It would be a really really cool contribution adding a lot of value to this project!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/apache/fineract/pull/651?email_source=notifications&email_token=AK5B5DHZOQX4H6OMQDMA2UTQYHUJ5A5CNFSM4JOK5LCKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEGVY3FQ#issuecomment-564891030>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AK5B5DHKBUQBQ3QAE7CM3ZTQYHUJ5ANCNFSM4JOK5LCA>.

mohitsinha commented:
@vorburger @xurror I was wondering some different deployment options for this task.

- Just restrict this to deploying the fineract backend and not worrying about the mysql instance. 
   The instance details can be passed via a ConfigMap, secret or environment variables.

- Keep the mysql container as a sidecar container alongside fineract container in the same pod.
 
I prefer the first one as it gives the user fine control over where the Mysql is hosted ( cloud, container etc).

mohitsinha commented on kubernetes/fineractmysql-deployment.yaml:
```diff
@@ -0,0 +1,101 @@
+  loadBalancer: {}
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
```
Correct me if I am wrong, I don't see values such as `storageClassName`, `PersistentVolume` etc.

xurror replied:
> Correct me if I am wrong, I don't see values such as `storageClassName`, `PersistentVolume` etc.

Yes, from what I saw declaring and deploying a PVC automatically creates a persistent storage

mohitsinha replied:
Sorry, if this question seems redundant. 
I just want to be sure. 
Is dynamic provisioning enabled by default in minikube.
https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/ 
I used the same yaml as yours and tried volume mounting a pod. 
It won't work unless this is enabled and that too with `storageClassName`.

If you are sure about this thing not causing any issues, then that's a go-ahead from me.
```
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  creationTimestamp: null
  labels:
    app: fineractmysql-claim0
  name: fineractmysql-claim0
spec:
  accessModes:
  - ReadOnlyMany
  resources:
    requests:
      storage: 100Mi
status: {}

vorburger commented:
@mohitsinha I quite liked that this includes a mySQL container, mostly because it could make it "ready to run", so that this isn't only a piece that somebody has to use as a starting point to put something together, but a fully functional, albeit certainly still demo and not prod, fully executable (think `kubectl apply` and it's up and running!) kind of thing. Let @xurror keep it? Just needs clean up.

mohitsinha commented:
@vorburger Can we have the mysql as a sidecar container rather than having a separate deployment for it?

I meant something along this line.
```
apiVersion: apps/v1
kind: Deployment
metadata:
  creationTimestamp: null
  labels:
    run: fineract
  name: fineract
spec:
  replicas: 1
  selector:
    matchLabels:
      run: fineract
  strategy: {}
  template:
    metadata:
      creationTimestamp: null
      labels:
        run: fineract
    spec:
      containers:
      - image: apache/fineract
        name: fineract
        resources: {}
      - image: mysql
        name: mysql
        resources: {}

status: {}

vorburger commented:
> @vorburger Can we have the mysql as a sidecar container rather than having a separate deployment for it?

@mohitsinha you technically could, as in I guess it would most likely also work, but I probably would not do that, in this case.. like, why would you? Sidecars are typically used for things which have to run in the same "Linux namespace", so that they share files and IP address and hostname, which can be useful for things like "envelopes" like "proxies" which only accept connections from localhost, or some log file scraper, or whatever. A completely separately running database wouldn't be a typical use case for a sidecar, I think, so the way @xurror is proposing it seems fine (and more "standard"), to me.

mohitsinha commented:
Yes. I agree with your opinion. @xurror Let's try to use standard Docker images for this one.

xurror commented:
One of my biggest challenge with this setting up the 2 databases needed and set them as data source for the backend.
It's like one always overwrites the previously declared datasource.

xurror commented:
So I need a little clarification, does the server automatically create the mifostenants-default database if it is not present?

mohitsinha commented:
No. it does not create the database by default.
We need `mifosplatform-tenants` and `mifostenants-{tenantName}`.
We also need the bare minimum records for the tenant inside `mifosplatform-tenants` to get the server running.

xurror commented:
I'll try to make it work with the 2 databases

mohitsinha commented:
I was just thinking that being able to use a persistent volume `hostPath` or a volume mount for Mysql data.
It would be handy for users, who restart the pod as and when needed. 

@vorburger @xurror What do you think?

vorburger commented:
Sounds good to me (but could also be a future improvement, after this PR).

xurror commented:
With this new update, I have this error:
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:294)
        at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:428)
Caused by: org.apache.catalina.LifecycleException: No UserDatabase component found under key UserDatabase
        at org.apache.catalina.realm.UserDatabaseRealm.startInternal(UserDatabaseRealm.java:262)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        ... 17 more


I'm not sure what causes it. @vorburger can provide assistance for this?

vorburger commented:
It most probably just means that it failed to connect to the database.
Check your log to find out the root cause and see if it's wrong hostname
("can't be reached") or bad uid/pwd.

> .
>

xurror commented:
From what I've seen from the docker-compose.yaml file, the docker instance only creates the databases but does not populate them with sample data(Well atleast I think that's what it does) but I have the feeling that the server needs sample data in the DB inorder to connect/load properly. Not sure if I'm right about this

cnangeh commented on README.md:
```diff
@@ -102,6 +102,26 @@ _(Note that in previous versions, the `mysqlserver` environment variable used at
+
+1. `git clone https://github.com/apache/fineract.git ; cd fineract`
+1. `minikube start`
+1. `kubectl apply -f kubernetes/`
```
you don't want to deploy all simultaneously.@xurror You would want the MySQL server up and running before starting up the fineract server. Unfortunately, the isn't a way to specify that with Kubernetes to the best of my knowledge. But basically you shouldn't start them up simultaneously.

cnangeh commented on kubernetes/fineract-server-deployment.yaml:
```diff
@@ -0,0 +1,66 @@
+        - name: MYSQL_ROOT_PASSWORD
+          value: mysql
+        - name: FINERACT_DEFAULT_TENANTDB_HOSTNAME
+          value: fineractmysql
```
Here you would want to specify the IP address of your MySQL cluster. You are using `fineractmysql` which doesn't make a lot of sense to me cause I don't see where you assign an IP address to it.

cnangeh commented on kubernetes/fineract-server-deployment.yaml:
```diff
@@ -0,0 +1,66 @@
+        - name: FINERACT_DEFAULT_TENANTDB_PORT
+          value: ""
+        - name: JAVA_OPTS
+          value: -Dfineract_tenants_url=jdbc:mysql:thin://fineractmysql/mifosplatform-tenants -Djava.awt.headless=true -XX:+UseG1GC -Dfile.encoding=UTF-8
```
Same thing here. Replace the 'fineractmysql' with the IP of your MySQL cluster

cnangeh commented:
@xurror The error message you are getting is because the fineract server is not connecting to the MySQL server. 
- Either MySQL is not running, so make sure that pod is running
or 
- the database doesn't exist on the MySQL server. Make sure the database exist on the MySQL server. it doesn't have to be prepopulated with data for the fineract server to start-up successfully.
or 
- the MySQL configuration you provide to the fineract server is wrong.

mohitsinha commented:
Have you created these two databases `mifosplatform-tenants` & `mifostenant-default`?
The `mifosplatform-tenants` should also have some records populated.
Please follow this https://github.com/openMF/mifosx/wiki/MySQL-Database-Setup

xurror commented:
Yes, I have the databases but no populated data.
I'll populate the databases and see how it goes

xurror commented:
This deployment should be good to go once the problem with the docker container is resolved.
It turns out the failure to connect to DB_HOST comes from the docker configuration of fineract-server

xurror commented:
I'm currently working on trying to resolve that too

mohitsinha commented:
@xurror I have some ideas about these issues. 
Just to be sure, inside the table`mifosplatform-tenants.tenant_server_connections` the `schema_server` should also be  `fineractmysql`.

xurror commented:
I have this error

Welcome to the Bitnami tomcat container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-tomcat
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-tomcat/issues

nami    INFO  Initializing tomcat
tomcat  INFO 
tomcat  INFO  ########################################################################
tomcat  INFO   Installation parameters for tomcat:
tomcat  INFO     Username: user
tomcat  INFO     Password: Not set during installation
tomcat  INFO   (Passwords are not shown for security reasons)
tomcat  INFO  ########################################################################
tomcat  INFO 
nami    INFO  tomcat successfully initialized
INFO  ==> Starting tomcat... 
Feb 10, 2020 4:31:09 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
INFO: The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
Feb 10, 2020 4:31:09 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-bio-8080"]
Feb 10, 2020 4:31:09 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-bio-8443"]
Feb 10, 2020 4:31:09 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["ajp-bio-8009"]
Feb 10, 2020 4:31:09 AM org.apache.catalina.startup.Catalina load
INFO: Initialization processed in 700 ms
Feb 10, 2020 4:31:10 AM org.apache.catalina.core.StandardService startInternal
INFO: Starting service Catalina
Feb 10, 2020 4:31:10 AM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/7.0.94
Feb 10, 2020 4:31:10 AM org.apache.catalina.realm.UserDatabaseRealm startInternal
SEVERE: Exception looking up UserDatabase under key UserDatabase
javax.naming.NameNotFoundException: Name [UserDatabase] is not bound in this Context. Unable to find [UserDatabase].
        at org.apache.naming.NamingContext.lookup(NamingContext.java:825)
        at org.apache.naming.NamingContext.lookup(NamingContext.java:173)
        at org.apache.catalina.realm.UserDatabaseRealm.startInternal(UserDatabaseRealm.java:253)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.realm.CombinedRealm.startInternal(CombinedRealm.java:236)
        at org.apache.catalina.realm.LockOutRealm.startInternal(LockOutRealm.java:120)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1227)
        at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:300)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.core.StandardService.startInternal(StandardService.java:444)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:758)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.startup.Catalina.start(Catalina.java:711)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:294)
        at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:428)

Feb 10, 2020 4:31:10 AM org.apache.catalina.realm.CombinedRealm startInternal
SEVERE: Failed to start "org.apache.catalina.realm.UserDatabaseRealm/1.0" realm
org.apache.catalina.LifecycleException: Failed to start component [Realm[UserDatabaseRealm]]
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:162)
        at org.apache.catalina.realm.CombinedRealm.startInternal(CombinedRealm.java:236)
        at org.apache.catalina.realm.LockOutRealm.startInternal(LockOutRealm.java:120)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1227)
        at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:300)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.core.StandardService.startInternal(StandardService.java:444)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:758)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        at org.apache.catalina.startup.Catalina.start(Catalina.java:711)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:294)
        at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:428)
Caused by: org.apache.catalina.LifecycleException: No UserDatabase component found under key UserDatabase
        at org.apache.catalina.realm.UserDatabaseRealm.startInternal(UserDatabaseRealm.java:262)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
        ... 17 more

Feb 10, 2020 4:31:10 AM org.apache.catalina.startup.HostConfig deployWAR
INFO: Deploying web application archive /bitnami/tomcat/data/fineract-provider.war



This error is seems to be similar to this. Don't know if I'm mistaken somewhere but there is no "UserDatabase" resource specified in our sever.xml

xurror commented:
I have this error

Welcome to the Bitnami tomcat container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-tomcat
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-tomcat/issues

nami INFO Initializing tomcat
tomcat INFO
tomcat INFO ########################################################################
tomcat INFO Installation parameters for tomcat:
tomcat INFO Username: user
tomcat INFO Password: Not set during installation
tomcat INFO (Passwords are not shown for security reasons)
tomcat INFO ########################################################################
tomcat INFO
nami INFO tomcat successfully initialized
INFO ==> Starting tomcat...
Feb 10, 2020 4:31:09 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
INFO: The APR based Apache Tomcat Native library which allows optimal performance in production environments was not found on the java.library.path: /usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
Feb 10, 2020 4:31:09 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-bio-8080"]
Feb 10, 2020 4:31:09 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["http-bio-8443"]
Feb 10, 2020 4:31:09 AM org.apache.coyote.AbstractProtocol init
INFO: Initializing ProtocolHandler ["ajp-bio-8009"]
Feb 10, 2020 4:31:09 AM org.apache.catalina.startup.Catalina load
INFO: Initialization processed in 700 ms
Feb 10, 2020 4:31:10 AM org.apache.catalina.core.StandardService startInternal
INFO: Starting service Catalina
Feb 10, 2020 4:31:10 AM org.apache.catalina.core.StandardEngine startInternal
INFO: Starting Servlet Engine: Apache Tomcat/7.0.94
Feb 10, 2020 4:31:10 AM org.apache.catalina.realm.UserDatabaseRealm startInternal
SEVERE: Exception looking up UserDatabase under key UserDatabase
javax.naming.NameNotFoundException: Name [UserDatabase] is not bound in this Context. Unable to find [UserDatabase].
at org.apache.naming.NamingContext.lookup(NamingContext.java:825)
at org.apache.naming.NamingContext.lookup(NamingContext.java:173)
at org.apache.catalina.realm.UserDatabaseRealm.startInternal(UserDatabaseRealm.java:253)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.realm.CombinedRealm.startInternal(CombinedRealm.java:236)
at org.apache.catalina.realm.LockOutRealm.startInternal(LockOutRealm.java:120)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1227)
at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:300)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.core.StandardService.startInternal(StandardService.java:444)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:758)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.startup.Catalina.start(Catalina.java:711)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:294)
at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:428)

Feb 10, 2020 4:31:10 AM org.apache.catalina.realm.CombinedRealm startInternal
SEVERE: Failed to start "org.apache.catalina.realm.UserDatabaseRealm/1.0" realm
org.apache.catalina.LifecycleException: Failed to start component [Realm[UserDatabaseRealm]]
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:162)
at org.apache.catalina.realm.CombinedRealm.startInternal(CombinedRealm.java:236)
at org.apache.catalina.realm.LockOutRealm.startInternal(LockOutRealm.java:120)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:1227)
at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:300)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.core.StandardService.startInternal(StandardService.java:444)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:758)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
at org.apache.catalina.startup.Catalina.start(Catalina.java:711)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:294)
at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:428)
Caused by: org.apache.catalina.LifecycleException: No UserDatabase component found under key UserDatabase
at org.apache.catalina.realm.UserDatabaseRealm.startInternal(UserDatabaseRealm.java:262)
at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:145)
... 17 more

Feb 10, 2020 4:31:10 AM org.apache.catalina.startup.HostConfig deployWAR
INFO: Deploying web application archive /bitnami/tomcat/data/fineract-provider.war

This error is seems to be similar to this. Don't know if I'm mistaken somewhere but there is no "UserDatabase" resource specified in our sever.xml

Commit:
FINERACT-783
 - Update README.md with instructions for kubernetes
 - Setup fineractmysql configuration file
 - Setup fineract-server configuration file
closes https://issues.apache.org/jira/browse/FINERACT-783
